### PR TITLE
Announce external links to screen readers

### PR DESCRIPTION
**Pull request by helaoutar**
Signed-off-by: Hamza El Aoutar <elaoutarhamza@gmail.com>

## Hey, I just made a Pull Request  which fixes [11102](https://github.com/backstage/backstage/issues/11102)

Changes summary: I added a visually hidden text (`, Open in a new tab`) inside external links

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/witty-readers-give.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/core-components': minor
```
Comment (benjdlambert) [inline .changeset/witty-readers-give.md:2]: This should be a `patch` release here, as `core-components` is still `0.x`

**Review by benjdlambert — COMMENTED** [review]
Thanks for this! Just one note about the changeset! :pray:

File: .changeset/witty-readers-give.md (hunk 1 above)
Comment (helaoutar) [reply .changeset/witty-readers-give.md]: Done!
Thanks

**Review by benjdlambert — APPROVED** [review]
Thank you!

**Pull request merged**